test/integration/driver_install_or_update_test.go

```diff
@@ -173,3 +174,85 @@ func TestHyperKitDriverInstallOrUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestHyperKitDriverSkipUpgrade(t *testing.T) {
+	if runtime.GOOS != "darwin" {
+		t.Skip("Skip if not darwin.")
+	}
+	MaybeParallel(t)
+	tests := []struct {
+		name             string
+		path             string
+		downloadExpected bool
+	}{
+		{
+			name:             "upgrade-v1.11.0-to-current",
+			path:             filepath.Join(*testdataDir, "hyperkit-driver-version-1.11.0"),
+			downloadExpected: false,
+		},
+		{
+			name:             "upgrade-v1.2.0-to-current",
+			path:             filepath.Join(*testdataDir, "hyperkit-driver-older-version"),
+			downloadExpected: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			originalPath := os.Getenv("PATH")
+			defer func() {
+				if err := os.Setenv("PATH", originalPath); err != nil {
+					t.Errorf("failed to restore PATH: %v", err)
+				}
+			}()
+
+			dir, err := ioutil.TempDir("", tc.name)
+			if err != nil {
+				t.Fatalf("Expected to create tempdir. test: %s, got: %v", tc.name, err)
+			}
+			defer func() {
+				if err := os.RemoveAll(dir); err != nil {
+					t.Errorf("Failed to remove dir %q: %v", dir, err)
+				}
+			}()
+
+			pwd, err := os.Getwd()
+			if err != nil {
+				t.Fatalf("Error not expected when getting working directory. test: %s, got: %v", tc.name, err)
+			}
+
+			driverPath := filepath.Join(pwd, tc.path, "docker-machine-driver-hyperkit")
+			if _, err = os.Stat(driverPath); err != nil {
+				t.Fatalf("Expected driver to exist. test: %s, got: %v", tc.name, err)
+			}
+
+			// change permission to allow driver to be executable
+			if err = os.Chmod(driverPath, 0700); err != nil {
+				t.Fatalf("Expected not expected when changing driver permission. test: %s, got: %v", tc.name, err)
+			}
+
+			if err = os.Setenv("PATH", filepath.Dir(driverPath)); err != nil {
```

Comment:
instead of setting Env variable for the whole test (that could conflict the other parallel tests) 
consider using the env only for the Comand
for example
```
	cmd.Env = append(os.Environ(), "DOCKER_HOST=/does/not/exist")
	if _, err := Run(t, cmd); err != nil {
//...
	}

```
here is an example here https://github.com/medyagh/minikube/blob/32922f4184c31fb405fb8dab7a1d7a8ef120b47a/test/integration/aaa_download_only_test.go#L170

Reply:
Done